packages/aws-cdk/lib/api/hotswap/lambda-functions.ts

```diff
@@ -1,11 +1,14 @@
 import { Writable } from 'stream';
-import * as archiver from 'archiver';
 import * as AWS from 'aws-sdk';
 import { flatMap } from '../../util';
 import { ISDK } from '../aws-auth';
 import { CfnEvaluationException, EvaluateCloudFormationTemplate } from '../evaluate-cloudformation-template';
 import { ChangeHotswapImpact, ChangeHotswapResult, HotswapOperation, HotswappableChangeCandidate } from './common';
 
+// namespace object imports won't work in the bundle for function exports
+// eslint-disable-next-line @typescript-eslint/no-require-imports
+const archiver = require('archiver');
```

Comment:
Does this become an error, or is this a silent failure? Can we add a linter rule? Can we fix this by turning on `esModuleInterop`?

Reply:
It becomes an `esbuild` warning which the tool treats as an error explicitly because of these failures. So we will detect these at build time. 

https://github.com/aws/aws-cdk/blob/ca79b7da453668f98e6d5fff5313207e87464476/tools/%40aws-cdk/node-bundle/src/api/bundle.ts#L336-L339

We could add `esModuleInterop` as well but seems like too big of a repo disturbance at this point.